checkstyle.sh

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
+
+# Check the code style of the UniMath project
+# Auke Booij, 2017
+
+#
+# START helper functions
+#
+
+die() {
+        echo "$@" >&2
+        exit 1
+}
+
+check_grep() {
+    typeset cmd="grep --color=always -E -A4 -B4 -n $1 $2"
+    typeset ret_code
+
+    output=$($cmd)
+    ret_code=$?
+
+    if [ $ret_code != 0 ] && [ $ret_code != 1 ]
+    then
+        die "Failed to run command:" "$cmd"
+    fi
+
+    if [[ ! -z $output ]]
+    then
+        (( FAILURES++ ))
+        echo "$output"
+    fi
+
+}
+
+check_command() {
+    check_grep "^[[:space:]]*($2)(\.|[^a-zA-Z][^.]*?\.)" "$1"
+}
+
+check_type() {
+    check_grep "[^[:space:]a-zA-Z.][^a-zA-Z.]+($2)($|[^a-zA-Z_])" "$1"
+}
+
+check_tactic() {
+    check_grep "(^|\.|;)[[:space:]]*($2)[^.;]*(\.|;)" "$1"
+}
+
+check_freestanding() {
+    check_grep "[^[:space:]](.*[^a-zA-Z]|)($2)\." "$1"
```

Comment:
`grep` gives an error here:
```
grep: empty (sub)expression
```
That's referring to the emptiness between `|` and `)`.

What did you have in mind there?

Reply:
It worked fine for me, so there seems to be disagreement among grep versions. Please try the new version of the style checker.